client/src/index.jsx

```diff
@@ -118,18 +118,34 @@ class App extends React.Component {
 		//Create "set" of animals
 		const animals = {};
 		sightings.forEach(element => {
-			if (animals[element.animal]) {
-				animals[element.animal].push(element.location);
+			const animal = element.animal;
```

Comment:
Won't this play poorly with the pagination?
I think you should foist this off onto the database

Reply:
Trying to think of how this would work.

Get a list of all the animals, search the db for each one? That seems like it could potentially scale poorly. Maybe build it as a searchOption?

Reply:
Subqueries maybe?